vita3k/ngs/src/modules/atrac9.cpp

```diff
@@ -182,10 +182,10 @@ bool Atrac9Module::decode_more_data(KernelState &kern, const MemState &mem, cons
             decoded_size -= skipped_samples;
         }
 
-        const uint32_t samples_left_after = (frame_bytes_gotten / superframe_size - 1) * samples_per_superframe;
+        const uint32_t samples_left_after = ((frame_bytes_gotten / superframe_size) - 1) * samples_per_superframe;
         if (bufparam.samples_discard_end_off > samples_left_after) {
             // last chunk
-            decoded_size -= bufparam.samples_discard_end_off;
+            decoded_size -= std::min(samples_per_superframe, bufparam.samples_discard_end_off - samples_left_after);
```

Comment:
If the previous `if` statement is executed, the bug will still occur.
```suggestion
            decoded_size -= std::min(decoded_size, bufparam.samples_discard_end_off - samples_left_after);
```